src/main/java/net/snowflake/ingest/utils/Utils.java

```diff
@@ -103,8 +103,29 @@ public static Properties createProperties(Properties inputProp) {
       properties.put(SFSessionProperty.PRIVATE_KEY.getPropertyKey(), parsePrivateKey(privateKey));
     }
 
-    if (!properties.containsKey(SFSessionProperty.PRIVATE_KEY.getPropertyKey())) {
-      throw new SFException(ErrorCode.MISSING_CONFIG, "private_key");
+    if (!properties.containsKey(Constants.AUTHORIZATION_TYPE)) {
+      properties.put(Constants.AUTHORIZATION_TYPE, Constants.JWT);
+    }
+
+    String authType = properties.get(Constants.AUTHORIZATION_TYPE).toString();
+    if (authType.equals(Constants.JWT)) {
+      if (!properties.containsKey(SFSessionProperty.PRIVATE_KEY.getPropertyKey())) {
+        throw new SFException(ErrorCode.MISSING_CONFIG, Constants.PRIVATE_KEY);
```

Comment:
No test for all these cases

Reply:
Added [here](https://github.com/snowflakedb/snowflake-ingest-java/blob/690c3b95fc0cc09a857cb244ce05b46b91e90dc2/src/test/java/net/snowflake/ingest/streaming/internal/OAuthBasicTest.java#L40)